docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc

```diff
@@ -0,0 +1,50 @@
+[[ml-nlp-search-compare]]
+= Search and compare text
+
+:keywords: {ml-init}, {stack}, {nlp}, text embedding
+:description: NLP tasks that allow you to search in unstructured text or \
+compare difference pieces of text.
+
+
+[discrete]
+[[ml-nlp-text-embedding]]
+== Text embedding
+
+Text embedding is a task which produces a mathematical representation of text 
+called an embedding. The {ml} model turns the text into an array of numerical 
+values (a vector). Pieces of content with similar meaning have similar 
+representation. This means it is possible to determine whether different pieces 
+of text are either semantically similar, different, or even opposite by using a 
+mathematical similarity function.
+
+This task is only responsible for producing the embedding alone. When the 
+embedding is created, it can be stored in a `dense_vector` field and used at 
+search time. Then you can apply the similarity function to an embedding 
```

Comment:
> Then you can apply the similarity function to an embedding representing a query and create semantic search capability.

This sentence is not clear to me. Perhaps this is better?

> For example, you can use these vectors in a k-nearest neighbor (kNN) search to achieve semantic search capabilities.

Reply:
We should link out to the new kNN docs: https://www.elastic.co/guide/en/elasticsearch/reference/master/knn-search.html

Reply:
I've added the link via e3b2698.